CHANGELOG.md

```diff
@@ -31,6 +31,17 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
   that describes them better, and they also behave similar to Mercurial's
   bookmarks.
 
+* `jj unsquash` has been deprecated in favor of `jj squash` and
+  `jj diffedit --restore-descendants`.
+
+  **Rationale:** `jj squash` can be used in interactive mode to pull
+  changes from one commit to another, including from a parent commit
+  to a child commit. For finer-grained application, such as when
```

Comment:
Nit: s/application/changes/g or "For fine-grained dependent diffs" but choose whatever you prefer.

Reply:
I like the second one. Done.

Reply:
Also, I got convinced for `diffedit --restore-descendants`, I even put it before `squash` in the deprecation message, it makes sense.